### PR TITLE
feat: Find record for client

### PR DESCRIPTION
**Pull request by maikelpoot**
This PR will add the functionality to get a vault record by record UUID without a group UUID

The `findForClient` uses the current client id as AccessibleByClient target for the search api.

Usage:
```golang

recordUUID, _ := uuid.Parse("00-00-00-00-00")

# We want secrets
additional := &model.VaultRecordAdditionalQueryParams{Secret: true, Audit: true}

# get vault record.
vaultRecord, _ = client.Vaults.FindByUUIDForClient(recordUUID, additional)
```

### CODE REVIEW

File: example/findrecord.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+	recordUUID, err := uuid.Parse(record)
+
+	var vaultRecord *model.VaultRecord
+
+	if err == nil {
+		vaultRecord, err = client.Vaults.FindByUUIDForClient(recordUUID, additional)
+	} else {
+		recordid := big.Int{}
+		recordid.SetString(record, 10)
```
Comment (hielkehoeve) [inline example/findrecord.go]: Hier kan je ook `strconv.ParseInt(record 10, 64)` gebruiken.

File: vaults.go [hunk 2]
```diff
@@ -102,6 +103,117 @@ func (s *VaultService) List(group *model.Group, query *model.VaultRecordQueryPar
...
+
+				/*
+					// The "good" way, make an extra call to retrieve the group
+					gs := newGroupService(s.sling)
+					group, err := gs.GetById(gid.Int64())
+					if err != nil {
+						err = fmt.Errorf("Could not retrieve group for VaultRecord. Error: %s", err.Error())
+					}
+					_ = group
+				*/
```
Comment (hielkehoeve) [inline vaults.go]: Mag weg toch?

File: vaults.go [hunk 3]
```diff
@@ -102,6 +103,117 @@ func (s *VaultService) List(group *model.Group, query *model.VaultRecordQueryPar
...
+
+	me := new(model.Client)
+
+	errorReport := new(model.ErrorReport)
+
+	_, err = s.sling.New().Get("/keyhub/rest/v1/client/me").Receive(&me, errorReport)
+	if err != nil {
+		err = fmt.Errorf("could not determine client details, error:", errorReport.Message)
+	}
+
```
Comment (hielkehoeve) [inline vaults.go:121]: Nog even een errorreport check

**Review by hielkehoeve — CHANGES_REQUESTED** [review]

File: example/findrecord.go (hunk 1 above)
Comment (maikelpoot) [reply example/findrecord.go]: fixed

File: vaults.go (hunk 3 above)
Comment (maikelpoot) [reply vaults.go:121]: fixed

File: vaults.go (hunk 2 above)
Comment (maikelpoot) [reply vaults.go]: done

**Review by hielkehoeve — APPROVED** [review]

**Pull request merged**